### PR TITLE
[docs][Expo config] Fix Constants.manifest reference and update verbiage

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Refs ENG-6588

# How

- By updating the reference of `Constants.manifest` to `Constants.expoConfig`. 
- Also, added a note for SDK 45 and below to use `Constants.manifest`.
- updated the overall verbiage, did some re-organization of text and added Callout components for blockquotes.
- Update link text in https://docs.expo.dev/versions/latest/config/app/ referring back to the `Configuration with app.json / app.config.js` page. Also, backported to SDK 45, 44, & 43.

# Test Plan

Run docs locally via `yarn dev` and then go open http://localhost:3002/workflow/configuration/.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/workflow/configuration.mdx [hunk 1]
```diff
@@ -106,11 +118,19 @@ module.exports = () => {
...
 ```
 
-To use this configuration with Expo CLI commands, set the environment variable either for specific commands or in your shell profile. To set environment variables for specific commands, prefix the command with the variables and values, for example: `MY_ENVIRONMENT=production eas update` (this is not anything unique to Expo CLI). On Windows you can approximate this with `npx cross-env MY_ENVIRONMENT=production eas update`, or use whichever other mechanism that you are comfortable with for environment variables.
+To use this configuration with Expo CLI commands, set the environment variable either for specific commands or in your shell profile. To set environment variables for specific commands, prefix the command with the variables and values as shown in the example:
+
+<Terminal cmd={['$ MY_ENVIRONMENT=production eas update']} />
+
+This is not anything unique to Expo CLI. On Windows you can approximate the above command with:
+
+<Terminal cmd={['$ npx cross-env MY_ENVIRONMENT=production eas update']} />
```
Comment (Simek) [inline docs/pages/workflow/configuration.mdx:127]:
```suggestion
<Terminal cmd={['$ npx cross-env MY_ENVIRONMENT=production eas update']} />
```

Hmm, `cross-env` is in maintenance mode only for almost 2 years now, maybe we should recommend an alternative package here? 
* https://github.com/kentcdodds/cross-env/issues/257

Comment (amandeepmittal) [reply docs/pages/workflow/configuration.mdx:127]: Thanks for catching that (wasn't aware of it). Will look into alternatives.

Comment (Simek) [reply docs/pages/workflow/configuration.mdx:127]: Let's stick to the `cross-env` for now, so we don't block this PR. 

I will check out few other approaches/alternative on my Windows machine later, and will report my findings to you, so we can decide how to handle that, and we can follow up with another PR.

Comment (amandeepmittal) [reply docs/pages/workflow/configuration.mdx:127]: That sounds good to me! Thanks for investigating for alternative 🙏

**Review by Simek — APPROVED** [review]
LGTM, thank for the updates!

**Pull request merged**